shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/watchers/DeploymentEventSource.java

```diff
@@ -0,0 +1,86 @@
+package com.redhat.service.bridge.shard.operator.watchers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.shard.operator.utils.LabelsBuilder;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.WatcherException;
+import io.javaoperatorsdk.operator.processing.event.AbstractEventSource;
+
+import static io.javaoperatorsdk.operator.processing.KubernetesResourceUtils.getUID;
+import static io.javaoperatorsdk.operator.processing.KubernetesResourceUtils.getVersion;
+
+/**
+ * Used by the controllers to watch changes on Deployment objects.
+ *
+ */
+public class DeploymentEventSource extends AbstractEventSource implements Watcher<Deployment> {
+    private static final Logger log = LoggerFactory.getLogger(DeploymentEventSource.class);
+
+    private final KubernetesClient client;
+
+    private final String applicationType;
+
+    public static DeploymentEventSource createAndRegisterWatch(KubernetesClient client, String applicationType) {
+        DeploymentEventSource deploymentEventSource = new DeploymentEventSource(client, applicationType);
+        deploymentEventSource.registerWatch(applicationType);
+        return deploymentEventSource;
+    }
+
+    private DeploymentEventSource(KubernetesClient client, String applicationType) {
+        this.client = client;
+        this.applicationType = applicationType;
+    }
+
+    private void registerWatch(String applicationType) {
+        client
+                .apps()
+                .deployments()
+                .inAnyNamespace()
+                .withLabels(
+                        new LabelsBuilder()
+                                .withManagedByOperator()
+                                .withApplicationType(applicationType)
+                                .build())
+                .watch(this);
+    }
+
+    @Override
+    public void eventReceived(Action action, Deployment deployment) {
+        log.info(
+                "Event received for action: {}, Deployment: {}",
```

Comment:
I think Rob mentioned something about using single quotes in these messages.

```suggestion
                "Event received for action: '{}', Deployment: '{}'",
```